test/client/http_service.test.js

```diff
@@ -0,0 +1,265 @@
+
+        expect(statusCode).toBe(200);
+        expect(body.toString()).toBe('POW!');
+      });
+
+      it('with options should work', async() => {
+        handleRequest = requestHandler;
+
+        const { statusCode, body } = await service.get({ url: TEST_SERVER_URL });
+
+        expect(statusCode).toBe(200);
+        expect(body.toString()).toBe('POW!');
+      });
+    });
+
+    describe('Submitting a POST request', () => {
+      const requestHandler = async(req, res) => {
+        const body = await readRequestBody(req);
+        res.writeHead(201, { 'Content-Type': 'text/plan' });
+        res.end(body);
+      };
+
+      it('with url should work', async() => {
+        handleRequest = requestHandler;
+
+        const { statusCode, body } = await service.post(TEST_SERVER_URL);
+
+        expect(statusCode).toBe(201);
+        expect(body.toString()).toBe('');
+      });
+
+      it('with options should work (raw body)', async() => {
+        handleRequest = requestHandler;
+
+        const requestBody = { foo: 'bar' };
+        const { statusCode, body } = await service.post({
+          url: TEST_SERVER_URL,
+          body: requestBody,
+          encodeBodyAsJSON: true
+        });
+
+        expect(statusCode).toBe(201);
+        expect(body.toString()).toBe(JSON.stringify(requestBody));
+      });
+
+      it('with options should work (stringified body)', async() => {
+        handleRequest = requestHandler;
+
+        const requestBody = JSON.stringify({ foo: 'bar' });
+        const { statusCode, body } = await service.post({
+          url: TEST_SERVER_URL,
+          body: requestBody
+        });
+
+        expect(statusCode).toBe(201);
+        expect(body.toString()).toBe(requestBody);
+      });
+    });
+
+    describe('Submitting a PUT request', () => {
+      const requestHandler = async(req, res) => {
+        const body = await readRequestBody(req);
+        res.writeHead(204, { 'Content-Type': 'text/plan' });
+        res.end(body);
+      };
+
+      it('with url should work', async() => {
+        handleRequest = requestHandler;
+
+        const { statusCode, body } = await service.put(TEST_SERVER_URL);
+
+        expect(statusCode).toBe(204);
+        expect(body).toBeFalsy();
+      });
+
+      it('with options should work (raw body)', async() => {
+        handleRequest = requestHandler;
+
+        const { statusCode, body } = await service.put({
+          url: TEST_SERVER_URL,
+          body: { foo: 'bar' },
+          encodeBodyAsJSON: true
+        });
+
+        expect(statusCode).toBe(204);
+        expect(body).toBeFalsy();
+      });
+
+      it('with options should work (stringified body)', async() => {
+        handleRequest = requestHandler;
+
+        const { statusCode, body } = await service.put({
+          url: TEST_SERVER_URL,
+          body: JSON.stringify({ foo: 'bar' })
+        });
+
+        expect(statusCode).toBe(204);
+        expect(body).toBeFalsy();
```

Comment:
should't we be doing more assertions in this test? the test server doesn't write back anything, so we haven't actually verified anything about if our outgoing data was correct or not (which i think is the intended purpose of this test, based on the description). Maybe the test server should echo the body back to the client, so that you can compare them here, which is what other tests are doing?